tests/tests.go

```diff
@@ -1316,11 +1317,17 @@ func (ctl *controller) waitBackup(databaseID, backupID int) error {
 		if backup == nil {
 			return errors.Errorf("backup %v for database %v not found", backupID, databaseID)
 		}
-		switch backup.Status {
-		case api.BackupStatusDone:
-			return nil
-		case api.BackupStatusFailed:
-			return errors.Errorf("backup %v for database %v failed", backupID, databaseID)
+		if shouldWait != nil {
```

Comment:
Instead of doing this, shall we implement a getBackup() method so that we don’t need to wait? Hacking the method interface isn't sound.

Reply:
We need to wait for the backup row status to be ARCHIVED.

Reply:
How about generalizing waitBackup() and provide a default implementation of shouldWait() using backup.Status?